### PR TITLE
Localization

### PR DESCRIPTION
**Pull request by SethGreylyn**
Here's a proof-of-concept for localization using [vue-i18n](https://github.com/kazupon/vue-i18n). There are some pros and cons to this library, and to the single-file approach currently adopted by the front-end.

PROS:

- By far the most-supported library for Vue
- Fairly easy to understand
- Lightweight and well-integrated

CONS:

- Soon to move to `intlify`, with an unknown migration scope
- Single-file approach is very brittle and repetitive (in particular, no clear way to share locales, making for a lot of repetition and a wide margin for error)

There are other patterns [in the docs](http://kazupon.github.io/vue-i18n/guide/component.html#shared-locale-messages-for-components) which I might explore, or we can decide to go forward with this approach in a "move fast" philosophy.

More generally, I'm a bit disenchanted with my first exposure to Vue, and am seriously considering rewriting the front-end in React. In my opinion it could be worth the initial pain. I'll keep thinking about it.

### CODE REVIEW

**Comment by SethGreylyn** [comment]
image

Here is a screenshot of the language-change tool. (I've only translated the headline of the `Über Uns` page, which is why the rest is all in German.) 

We can look into changing the displayed labels to flags and styling the dropdown better, but this represents the core functionality, which can be built upon by adding translations file by file. I'll focus on that for the next couple of hours.

**Comment by SethGreylyn** [comment]
The localization infrastructure is now in place in all the views and components (at least unless I missed anything). I did not provide very many English-language translations, but these can be done incrementally; note that German is the default fallback, so if an `"en"` dictionary is missing a key, the localization engine will use the `"de"` dictionary's entry; if the key does not exist, it will just display the key.

This pattern exposes some limitations of the current form split, namely with the repetition of many fields. There isn't an elegant way to avoid this repetition without refactoring the components, which I considered beyond the scope, but which we should address at some point in the future.

File: labshare-client/src/App.vue [hunk 1]
```diff
@@ -46,20 +67,21 @@
...
+      if (this.$store.state.profile.role) {
+        if (this.$store.state.profile.role === "human") {
+          return this.$store.state.profile.contact.firstname;
         } else {
-          return this.$store.state.profile.name
+          return this.$store.state.profile.name;
         }
       } else {
-        return "User"
+        return "User";
```
Comment (SethGreylyn) [inline labshare-client/src/App.vue]: Should probably be localized somehow, but pretty low-priority

File: labshare-client/src/components/HelperForm.vue [hunk 2]
```diff
@@ -1,89 +1,167 @@
...
         <b-form-checkbox-group
           id="skills"
           v-model="formData.details.skills"
-          :options="labSkills"
           name="name"
-          stacked></b-form-checkbox-group>
+          stacked
+        >
+          <b-form-checkbox
+            v-for="(skill, i) in labSkills"
```
Comment (SethGreylyn) [inline labshare-client/src/components/HelperForm.vue]: Now that I look at it again, `labSkills` could probably just be a string array, but it's not worth refactoring for its own sake (and arguably the object still has expressive value)

File: labshare-client/src/main.js [hunk 3]
```diff
@@ -1,23 +1,26 @@
-import Vue from 'vue'
-import Vuex from 'vuex'
-import router from './router.js'
-import store from './store'
-import App from './App.vue'
-import './http'
-import profile_load_mixin from './mixins/profile_fetch'
-import { BootstrapVue, IconsPlugin } from 'bootstrap-vue'
+import Vue from "vue";
```
Comment (SethGreylyn) [inline labshare-client/src/main.js]: My VSCode is apparently set to prefer double-quotes and automatically put in semicolons. I can can change these settings if that's an issue.

Comment (SethGreylyn) [reply labshare-client/src/main.js]: On the other hand, we should probably just use Prettier to automatically flatten these issues between different workstations

**Comment by SethGreylyn** [comment]
From a discussion on Discord:

> do you think we really need the Component based localization? Isn't it much better to have single localization file?

@Kavakuo  It depends on your architecture. In my full-time job we have two monolithic localization files (with the idea that we'll have authoritative translations that can be shared), and honestly after a certain size it becomes unwieldy and difficult to manually keep track and make sure that you're not duplicating information, just because you can't be sure (without an encyclopedic knowledge of the current state of the component architecture, or an awesome tool, or clever grepping) which components are using a translation, which components coincidentally have the same information but shouldn't share the same source, and which ones only look like they should share the same information at first glance but don't.

Component-based translations give you confidence that the translation is always going to be accurate, but it comes at the cost of needing more clever component architecture---it poses the same questions, but shifts the burden to the component tree, rather than the dev process.

NEW COMMENTS:

I'm not wedded to either approach, and in this case a single file would help reduce duplication much more cost-effectively than refactoring the component structure, so I'll give it a shot in a bit.

**Comment by Kavakuo** [comment]
Maybe it's best to have both methods working. Then we can put text that is used in multiple places in the large file and text that is used only in single components into the component file.

**Comment by SethGreylyn** [comment]
K, I have a central list in `main.js` for shared translations (which I experimented with earlier, but I misinterpreted the docs and so it didn't work...but a second look had me figuring it out). I've also added the componentwise translation to the new `Profile` component.

**Comment by SethGreylyn** [comment]
It should now just be a matter of typing to add English translations where they're necessary, but I'm out of types for the day. ;)

**Comment by SethGreylyn** [comment]
My only suggestion is to make `de` the fallback rather than `en`, at least given the current state of the strings, but that's a minor detail which we can fill out just by typing.

**Comment by Kavakuo** [comment]
Realized the same und pushed it straight to master :D

**Pull request merged**